### DISCUSSION TITLE
Nginx-Proxy-Manager Upgrade Guide

### DISCUSSION THREAD

**Discussion by jeffh2os** [NginxProxyManager/nginx-proxy-manager#3073] [Show and tell]
After not being able to upgrade with the basic docker-compose commands I have documented my upgrade method and figured may be useful to someone else out there.

Check Releases page:
https://github.com/NginxProxyManager/nginx-proxy-manager/releases

Stop Docker Containers: 
`for i in $(docker ps -a -q);do docker stop $i ;done`

Backup: 
`sudo tar cfJ /opt/nginx-backup-$(docker ps -a |grep nginx-proxy-manager|cut -d ':' -f2 |awk '{print $1}').tar.xz /opt/nginxproxymanager/*`

Pull down new images: 
`docker pull jc21/nginx-proxy-manager:x.xx.xx`

edit config: `sudo vim /opt/nginxproxymanager/docker-compose.yaml`
change line: image: 'jc21/nginx-proxy-manager:2.xx.xx'
to reflect new version number

Start Nginx Proxy Manager:
`docker-compose up -d`

follow logs on started up to make sure no errors
`docker logs --follow {image id}`

if you run into errors with:
error create table migrations (id int unsigned not null auto_increment primary key, name varchar(255), batch int, migration_time timestamp) - ER_CANT_CREATE_TABLE: Can't create table npm.migrations (errno: 13 "Permission denied")

Restart DB container:
`docker restart {image id}`

**Comment by NeuroDawg** [comment]
I'm glad I use Portainer. Updating was as simple as telling portainer to recreate the container and pull the new image when it does so. (using docker and portainer on a Ubuntu server)

**Comment by talesam** [comment]
I'm facing an issue. I'm currently on version 2.9.22, and to prevent encountering further problems, I've modified my docker-compose to prevent any updates. This was necessary because the last time I updated, I experienced several issues. Additionally, I'm using the older docker-compose configuration that employs separate containers for the database data. Is there a recommended approach for me to upgrade from version 2.9.* to 2.10?

**Reply by jeffh2os** [reply]
I believe I was on 2.9.xx as well and have two containers (nginxproxymanager_app_1 / nginxproxymanager_db_1) as well and the way I have above worked for me to go to 2.10.xx
